sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/HiveThriftServer2.scala

```diff
@@ -49,30 +49,32 @@ object HiveThriftServer2 extends Logging {
 
   /**
    * :: DeveloperApi ::
-   * Starts a new thrift server with the given context.
+   * Starts a new thrift server with the given SparkSession.
    *
-   * @param sqlContext SQLContext to use for the server
+   * @param sparkSession SparkSession to use for the server
    * @param exitOnError Whether to exit the JVM if HiveThriftServer2 fails to initialize. When true,
    *                    the call logs the error and exits the JVM with exit code -1. When false, the
    *                    call throws an exception instead.
    */
   @Since("4.0.0")
   @DeveloperApi
-  def startWithContext(sqlContext: SQLContext, exitOnError: Boolean): HiveThriftServer2 = {
+  def startWithSparkSession(
```

Comment:
It seems that aside from the unreleased API added in 4.0, this pr has not caused any breaking changes.

If so, the changes is fine to me. WDYT? @HyukjinKwon  @yaooqinn